### PR TITLE
Add log analytics workspace to Vault

### PR DESCRIPTION
**Pull request by jensone54**
## Description

Added log ananlytics workspace to key vault module

## References

[](https://app.zenhub.com/workspaces/mighty-ducks---the-outlaws-6193fe815d79fc0011e741b1/issues/energinet-datahub/geh-terraform-modules/24)

* #0000

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,4 +15,4 @@
 - [Azure Application Insights](./azure/application-insights)
 - [Azure Function App](./azure/function-app)
 - [Azure App Service Plan](./azure/app-service-plan)
-- [Azure Event Hub](./azure/eventhub-namespace)
```
Comment (djorgensendk) [inline README.md:18]: the line concerning Event Hub shouldn't be remove, instead a new line with Log Analytics workspace should be added

File: azure/key-vault/README.md [hunk 2]
```diff
@@ -31,8 +32,10 @@ This module creates the following resources.
 | `sku_name` | `string` | | **Required** | The Name of the SKU used for this Key Vault. Possible values are `standard` and `premium`. |
 | `access_policies` | `list` | `[]` | |  A list of objects describing the Key Vault access policies. See [Access Policy](#access-policy). |
 | `enabled_for_template_deployment` | `bool` | `false` | | Boolean flag to specify whether Azure Resource Manager is permitted to retrieve secrets from the key vault. Defaults to false. |
+| `log_analytics_workspace_id` | `string` | | **Required** | | Id of Log analytics workspace associated with the vault  |
```
Comment (djorgensendk) [inline azure/key-vault/README.md:35]: this attribute should be added to the example under ##Usage as well

File: azure/key-vault/main.tf [hunk 3]
```diff
@@ -67,4 +67,27 @@ resource "azurerm_key_vault_access_policy" "this" {
   key_permissions         = try(var.access_policies[count.index].key_permissions, [])
   certificate_permissions = try(var.access_policies[count.index].certificate_permissions, [])
   storage_permissions     = try(var.access_policies[count.index].storage_permissions, [])
+}
+
+resource "azurerm_monitor_diagnostic_setting" "this" {
+  name               = "vault-log-analytics-diagnostic-setting"
```
Comment (djorgensendk) [inline azure/key-vault/main.tf:73]: Please make sure that the identation is the same

File: azure/key-vault/main.tf [hunk 4]
```diff
@@ -67,4 +67,27 @@ resource "azurerm_key_vault_access_policy" "this" {
...
+  target_resource_id         = azurerm_key_vault.this.id
+  log_analytics_workspace_id = var.log_analytics_workspace_id
+
+  log {
+    category = "AuditEvent"
+    enabled  = false
```
Comment (djorgensendk) [inline azure/key-vault/main.tf:79]: This is the diagnostics log disabled?

File: azure/key-vault/variables.tf [hunk 5]
```diff
@@ -69,4 +69,9 @@ variable tags {
   type        = any
   description = "(Optional) A mapping of tags to assign to the resource."
   default     = {}
-}
\ No newline at end of file
+}
+
+variable log_analytics_workspace_id {
+  type = string
+  description = "(Required) The id of the log analytics workspace where the keyvault will log events (e.g. audit events)"
```
Comment (djorgensendk) [inline azure/key-vault/variables.tf:76]:
```suggestion
  description = "(Required) The ID of the Log Analytics workspace where the Key vault will log events (e.g. audit events)"
```

**Review by djorgensendk — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jensone54) [reply README.md:18]: obviously not mistake

File: azure/key-vault/README.md (hunk 2 above)
Comment (jensone54) [reply azure/key-vault/README.md:35]: actually it should go away will remove

File: azure/key-vault/main.tf [hunk 6]
```diff
@@ -67,4 +67,27 @@ resource "azurerm_key_vault_access_policy" "this" {
...
+
+    retention_policy {
+      enabled = false
+    }
```
[the remark this answers is not in the document]
Comment (jensone54) [reply azure/key-vault/main.tf:83]: is that not retention for a deleted vault?

File: azure/key-vault/README.md (hunk 2 above)
Comment (djorgensendk) [reply azure/key-vault/README.md]: I don't understand why this variable should be removed

Comment (jensone54) [reply azure/key-vault/README.md]: it shouldn't I thought I had an output variablee , this one is staying

File: azure/key-vault/main.tf (hunk 4 above)
Comment (jensone54) [reply azure/key-vault/main.tf:79]: sorry thought is was referring to a property in the vault itself

File: azure/key-vault/README.md (hunk 2 above)
Comment (jensone54) [reply azure/key-vault/README.md]: added to usage

**Review by djorgensendk — APPROVED** [review]

**Pull request merged**